bumble/profiles/ascs.py

```diff
@@ -0,0 +1,155 @@
+        self.id = id
+        self.characteristic = characteristic
+        self.characteristic.value = gatt.CharacteristicValue(read=self.on_read)
+
+    def on_read(self, connection: device.Connection) -> bytes:
+        return struct.pack('BB', self.id, self.state)
+
+
+# -----------------------------------------------------------------------------
+# Server
+# -----------------------------------------------------------------------------
+class Service(gatt.TemplateService):
+    UUID = gatt.GATT_COORDINATED_SET_IDENTIFICATION_SERVICE
+
+    sink_ase: List[AudioStreamEndpoint] = []
+    source_ase: List[AudioStreamEndpoint] = []
+
+    def __init__(self, num_sink_ase: int = 0, num_source_ase: int = 0) -> None:
+        characteristics = []
+
+        for i in range(1, num_sink_ase + 1):
+            characteristic = gatt.Characteristic(
+                uuid=gatt.GATT_SINK_ASE_CHARACTERISTIC,
+                properties=gatt.Characteristic.Properties.READ
+                | gatt.Characteristic.Properties.NOTIFY,
+                permissions=gatt.Characteristic.Permissions.READABLE,
+            )
+            self.sink_ase.append(
+                AudioStreamEndpoint(
+                    id=i,
+                    characteristic=characteristic,
+                )
+            )
+            characteristics.append(characteristic)
+
+        for i in range(1, num_source_ase + 1):
+            characteristic = gatt.Characteristic(
+                uuid=gatt.GATT_SOURCE_ASE_CHARACTERISTIC,
+                properties=gatt.Characteristic.Properties.READ
+                | gatt.Characteristic.Properties.NOTIFY,
+                permissions=gatt.Characteristic.Permissions.READABLE,
+            )
+            self.source_ase.append(
+                AudioStreamEndpoint(
+                    id=i,
+                    characteristic=characteristic,
+                )
+            )
+            characteristics.append(characteristic)
+        characteristics.append(
+            gatt.Characteristic(
+                uuid=gatt.GATT_ASE_CONTROL_POINT_CHARACTERISTIC,
+                properties=gatt.Characteristic.Properties.WRITE
+                | gatt.Characteristic.Properties.WRITE_WITHOUT_RESPONSE
+                | gatt.Characteristic.Properties.NOTIFY,
+                permissions=gatt.Characteristic.Permissions.READABLE
+                | gatt.Characteristic.Permissions.WRITEABLE,
+                value=gatt.CharacteristicValue(write=self.on_ase_control_point),
+            )
+        )
+
+        super().__init__(characteristics)
+
+    def on_ase_control_point(self, connection: device.Connection, data: bytes) -> None:
+        pass
+
+
+# -----------------------------------------------------------------------------
+# Client
+# -----------------------------------------------------------------------------
+class ServiceProxy(gatt_client.ProfileServiceProxy):
+    SERVICE_CLASS = Service
+
+    sirk: gatt_client.CharacteristicProxy
```

Comment:
I think this has been copy/pasted by mistake from another profile

Reply:
No, why do you think so?